### ISSUE TITLE
[dev.icinga.com #11755] icinga2.debug file not updated by API created objects

### ISSUE THREAD

**Issue by icinga-migration** [Icinga/icinga2#4200]
This issue has been migrated from Redmine: https://dev.icinga.com/issues/11755

**Created by mnardin on 2016-05-10 12:54:20 +00:00**

Assignee: _(none)_
Status: _New_ 
Target Version: _(none)_
Last Update: _2016-05-10 13:07:11 +00:00 (in Redmine)_

    Icinga Version: 2.4.7
    Backport?: Not yet backported
    Include in Changelog: 1

---

Hi,
I've added a downtime at 14:39:48 through the API. I was expecting to see the object listed in the core. The object was there only after a restart.

    [root@icingam01-p notifications]# date && icinga2 object list --name 'protst!*' --type Downtime
    Tue May 10 14:40:26 CEST 2016
    [root@icingam01-p notifications]# service icinga2 restart
    Checking configuration: Done
    Stopping Icinga 2: .Done
    Starting Icinga 2: Done
    [root@icingam01-p notifications]# date && icinga2 object list --name 'protst!*' --type Downtime
    Tue May 10 14:40:58 CEST 2016
    Object 'protst!icingam01-p-1462883988-0' of type 'Downtime':
      % declared in '/var/lib/icinga2/api/packages/_api/icingam01-p-1453216567-1/conf.d/downtimes/protst!icingam01-p-1462883988-0.conf', lines 1:0-1:57
      * __name = "protst!icingam01-p-1462883988-0"
      * author = "pulpy"
        % = modified in '/var/lib/icinga2/api/packages/_api/icingam01-p-1453216567-1/conf.d/downtimes/protst!icingam01-p-1462883988-0.conf', lines 2:2-2:17
      * comment = "test"
        % = modified in '/var/lib/icinga2/api/packages/_api/icingam01-p-1453216567-1/conf.d/downtimes/protst!icingam01-p-1462883988-0.conf', lines 3:2-3:17
      * config_owner = ""
        % = modified in '/var/lib/icinga2/api/packages/_api/icingam01-p-1453216567-1/conf.d/downtimes/protst!icingam01-p-1462883988-0.conf', lines 4:2-4:18
      * duration = 0
        % = modified in '/var/lib/icinga2/api/packages/_api/icingam01-p-1453216567-1/conf.d/downtimes/protst!icingam01-p-1462883988-0.conf', lines 5:2-5:20
      * end_time = 1462884240
        % = modified in '/var/lib/icinga2/api/packages/_api/icingam01-p-1453216567-1/conf.d/downtimes/protst!icingam01-p-1462883988-0.conf', lines 6:2-6:29
      * fixed = true
        % = modified in '/var/lib/icinga2/api/packages/_api/icingam01-p-1453216567-1/conf.d/downtimes/protst!icingam01-p-1462883988-0.conf', lines 7:2-7:13
      * host_name = "protst"
        % = modified in '/var/lib/icinga2/api/packages/_api/icingam01-p-1453216567-1/conf.d/downtimes/protst!icingam01-p-1462883988-0.conf', lines 8:2-8:38
      * name = "icingam01-p-1462883988-0"
      * package = "_api"
      * scheduled_by = ""
        % = modified in '/var/lib/icinga2/api/packages/_api/icingam01-p-1453216567-1/conf.d/downtimes/protst!icingam01-p-1462883988-0.conf', lines 9:2-9:18
      * service_name = ""
      * start_time = 1462883940
        % = modified in '/var/lib/icinga2/api/packages/_api/icingam01-p-1453216567-1/conf.d/downtimes/protst!icingam01-p-1462883988-0.conf', lines 10:2-10:31
      * templates = [ "icingam01-p-1462883988-0" ]
        % = modified in '/var/lib/icinga2/api/packages/_api/icingam01-p-1453216567-1/conf.d/downtimes/protst!icingam01-p-1462883988-0.conf', lines 1:0-1:57
      * triggered_by = ""
        % = modified in '/var/lib/icinga2/api/packages/_api/icingam01-p-1453216567-1/conf.d/downtimes/protst!icingam01-p-1462883988-0.conf', lines 11:2-11:18
      * type = "Downtime"
      * zone = "DR"
        % = modified in '/var/lib/icinga2/api/packages/_api/icingam01-p-1453216567-1/conf.d/downtimes/protst!icingam01-p-1462883988-0.conf', lines 13:2-13:12

debug.log extract:

    [root@icingam01-p notifications]# grep protst /var/log/icinga2/debug.log|egrep -v "notice/Dependency"
    [2016-05-10 14:39:34 +0200] debug/HostDbObject: host dependencies for 'protst'
    [2016-05-10 14:39:34 +0200] debug/HostDbObject: host contacts: protst
    [2016-05-10 14:39:34 +0200] debug/HostDbObject: host contactgroups: protst
    [2016-05-10 14:39:35 +0200] debug/ServiceDbObject: service dependencies for 'protst!ping'
    [2016-05-10 14:39:35 +0200] debug/ServiceDbObject: service contacts: protst!ping
    [2016-05-10 14:39:35 +0200] debug/ServiceDbObject: service contactgroups: protst!ping
    [2016-05-10 14:39:35 +0200] debug/ServiceDbObject: service dependencies for 'protst!CPU'
    [2016-05-10 14:39:35 +0200] debug/ServiceDbObject: service parents: protst!SNMP-Status
    [2016-05-10 14:39:35 +0200] debug/ServiceDbObject: service contacts: protst!CPU
    [2016-05-10 14:39:35 +0200] debug/ServiceDbObject: service contactgroups: protst!CPU
    [2016-05-10 14:39:35 +0200] debug/ServiceDbObject: service dependencies for 'protst!memory-win'
    [2016-05-10 14:39:35 +0200] debug/ServiceDbObject: service parents: protst!SNMP-Status
    [2016-05-10 14:39:35 +0200] debug/ServiceDbObject: service contacts: protst!memory-win
    [2016-05-10 14:39:35 +0200] debug/ServiceDbObject: service contactgroups: protst!memory-win
    [2016-05-10 14:39:35 +0200] debug/ServiceDbObject: service dependencies for 'protst!disk-state'
    [2016-05-10 14:39:35 +0200] debug/ServiceDbObject: service parents: protst!SNMP-Status
    [2016-05-10 14:39:35 +0200] debug/ServiceDbObject: service contacts: protst!disk-state
    [2016-05-10 14:39:35 +0200] debug/ServiceDbObject: service contactgroups: protst!disk-state
    [2016-05-10 14:39:35 +0200] debug/ServiceDbObject: service dependencies for 'protst!time'
    [2016-05-10 14:39:35 +0200] debug/ServiceDbObject: service parents: protst!SNMP-Status
    [2016-05-10 14:39:35 +0200] debug/ServiceDbObject: service contacts: protst!time
    [2016-05-10 14:39:35 +0200] debug/ServiceDbObject: service contactgroups: protst!time
    [2016-05-10 14:39:36 +0200] debug/ServiceDbObject: service dependencies for 'protst!SNMP-Status'
    [2016-05-10 14:39:36 +0200] debug/ServiceDbObject: service contacts: protst!SNMP-Status
    [2016-05-10 14:39:36 +0200] debug/ServiceDbObject: service contactgroups: protst!SNMP-Status
    [2016-05-10 14:39:40 +0200] debug/IdoMysqlConnection: Query: UPDATE icinga_hosts SET action_url = '',  active_checks_enabled = '1',  address = '10.139.0.154',  address6 = '',  alias = 'protst',  check_command_args = '',  check_command_object_id = 23,  check_interval = '1',  config_type = '1',  display_name = 'protst',  event_handler_enabled = '1',  failure_prediction_enabled = '0',  flap_detection_enabled = '0',  freshness_checks_enabled = '1',  freshness_threshold = '60',  high_flap_threshold = '30',  host_object_id = 1415,  icon_image = '',  icon_image_alt = '',  instance_id = 1,  low_flap_threshold = '30',  max_check_attempts = '3',  notes = '',  notes_url = '',  notification_interval = '0',  notifications_enabled = '1',  notify_on_down = '0',  notify_on_downtime = '1',  notify_on_flapping = '1',  notify_on_recovery = '1',  notify_on_unreachable = '0',  obsess_over_host = '0',  passive_checks_enabled = '1',  process_performance_data = '1',  retain_nonstatus_information = '1',  retain_status_information = '1',  retry_interval = '1' WHERE host_object_id = 1415
    [2016-05-10 14:39:48 +0200] debug/HttpRequest: line: POST /v1/actions/schedule-downtime?type=Host&filter=host.name==%22protst%22 HTTP/1.1, tokens: 3
    [2016-05-10 14:39:48 +0200] information/HttpServerConnection: Request: POST /v1/actions/schedule-downtime?filter[]=host.name==%22protst%22&type[]=Host (director)
    [2016-05-10 14:39:48 +0200] information/ConfigCompiler: Compiling config file: /var/lib/icinga2/api/packages/_api/icingam01-p-1453216567-1/conf.d/downtimes/protst!icingam01-p-1462883988-0.conf
    [2016-05-10 14:39:48 +0200] notice/Downtime: Added downtime 'protst!icingam01-p-1462883988-0' between '2016-05-10 14:39:00' and '2016-05-10 14:44:00'.
    [2016-05-10 14:39:48 +0200] notice/ApiListener: Received update for object: {"config":"object Downtime \"icingam01-p-1462883988-0\" ignore_on_error {\n\tauthor = \"pulpy\"\n\tcomment = \"test\"\n\tconfig_owner = \"\"\n\tduration = 0.000000\n\tend_time = 1462884240.000000\n\tfixed = true\n\thost_name = \"protst\"\n\tscheduled_by = \"\"\n\tstart_time = 1462883940.000000\n\ttriggered_by = \"\"\n\tversion = 1462883988.366245\n\tzone = \"DR\"\n}\n","modified_attributes":{},"name":"protst!icingam01-p-1462883988-0","original_attributes":[],"type":"Downtime","version":1462883988.3662450314}
    [2016-05-10 14:39:48 +0200] warning/ApiListener: Ignoring config update for object 'protst!icingam01-p-1462883988-0' of type 'Downtime'. 'api' does not accept config.
    [2016-05-10 14:39:53 +0200] debug/IdoMysqlConnection: Query: INSERT INTO icinga_scheduleddowntime (author_name, comment_data, downtime_type, duration, endpoint_object_id, entry_time, instance_id, internal_downtime_id, is_fixed, is_in_effect, name, object_id, scheduled_end_time, scheduled_start_time, was_started) VALUES ('pulpy', 'test', '2', '0', 5, FROM_UNIXTIME(1462883988), 1, '908', '1', '1', 'protst!icingam01-p-1462883988-0', 1415, FROM_UNIXTIME(1462884240), FROM_UNIXTIME(1462883940), '1')
    [2016-05-10 14:39:53 +0200] debug/IdoMysqlConnection: Query: INSERT INTO icinga_downtimehistory (author_name, comment_data, downtime_type, duration, endpoint_object_id, entry_time, instance_id, internal_downtime_id, is_fixed, is_in_effect, name, object_id, scheduled_end_time, scheduled_start_time, was_started) VALUES ('pulpy', 'test', '2', '0', 5, FROM_UNIXTIME(1462883988), 1, '908', '1', '1', 'protst!icingam01-p-1462883988-0', 1415, FROM_UNIXTIME(1462884240), FROM_UNIXTIME(1462883940), '1')
    [2016-05-10 14:39:59 +0200] debug/DbEvents: Updating reachability for checkable 'protst!SNMP-Status':  reachable for 4 children.
    [2016-05-10 14:39:59 +0200] debug/DbEvents: Updating reachability for checkable 'protst!CPU':  reachable.
    [2016-05-10 14:39:59 +0200] debug/DbEvents: Updating reachability for checkable 'protst!disk-state':  reachable.
    [2016-05-10 14:39:59 +0200] debug/DbEvents: Updating reachability for checkable 'protst!time':  reachable.
    [2016-05-10 14:39:59 +0200] debug/DbEvents: Updating reachability for checkable 'protst!memory-win':  reachable.
    [2016-05-10 14:39:59 +0200] debug/DbEvents: add checkable check history for 'protst!SNMP-Status'
    [2016-05-10 14:40:14 +0200] debug/DbEvents: add checkable check history for 'protst'
    [2016-05-10 14:40:14 +0200] information/Checkable: Checking for configured notifications for object 'protst'
    [2016-05-10 14:40:14 +0200] debug/Checkable: Checkable 'protst' has 2 notification(s).
    [2016-05-10 14:40:14 +0200] notice/Notification: Attempting to send notifications for notification object 'protst!mail-to-host-owner'.
    [2016-05-10 14:40:14 +0200] debug/DbEvents: add notification history for 'protst'
    [2016-05-10 14:40:14 +0200] notice/Notification: Attempting to send notifications for notification object 'protst!mail-to-unarm'.
    [2016-05-10 14:40:14 +0200] information/Notification: Sending notification 'protst!mail-to-unarm' for user 'unarm'
    [2016-05-10 14:40:14 +0200] debug/DbEvents: add notification history for 'protst'
    [2016-05-10 14:40:14 +0200] debug/DbEvents: add contact notification history for service 'protst' and user 'unarm'.
    [2016-05-10 14:40:14 +0200] debug/DbEvents: add log entry history for 'protst'
    [2016-05-10 14:40:14 +0200] information/Notification: Completed sending notification 'protst!mail-to-unarm' for checkable 'protst' and user 'unarm'.
    [2016-05-10 14:40:14 +0200] debug/DbEvents: add checkable check history for 'protst!ping'
    [2016-05-10 14:40:14 +0200] debug/IdoMysqlConnection: Query: INSERT INTO icinga_logentries (endpoint_object_id, entry_time, entry_time_usec, instance_id, logentry_data, logentry_time, logentry_type, object_id) VALUES (5, FROM_UNIXTIME(1462884014), '473582', 1, 'HOST NOTIFICATION: unarm;protst;DOWNTIMESTART (UP);hostalive;PING OK - Packet loss = 0%, RTA = 6.89 ms', FROM_UNIXTIME(1462884014), '524288', 1415)
    [2016-05-10 14:40:54 +0200] debug/HostDbObject: host dependencies for 'protst'
    [2016-05-10 14:40:54 +0200] debug/HostDbObject: host contacts: protst
    [2016-05-10 14:40:54 +0200] debug/HostDbObject: host contactgroups: protst
    [2016-05-10 14:40:55 +0200] debug/ServiceDbObject: service dependencies for 'protst!ping'
    [2016-05-10 14:40:55 +0200] debug/ServiceDbObject: service contacts: protst!ping
    [2016-05-10 14:40:55 +0200] debug/ServiceDbObject: service contactgroups: protst!ping
    [2016-05-10 14:40:55 +0200] debug/ServiceDbObject: service dependencies for 'protst!CPU'
    [2016-05-10 14:40:55 +0200] debug/ServiceDbObject: service parents: protst!SNMP-Status
    [2016-05-10 14:40:55 +0200] debug/ServiceDbObject: service contacts: protst!CPU
    [2016-05-10 14:40:55 +0200] debug/ServiceDbObject: service contactgroups: protst!CPU
    [2016-05-10 14:40:55 +0200] debug/ServiceDbObject: service dependencies for 'protst!memory-win'
    [2016-05-10 14:40:55 +0200] debug/ServiceDbObject: service parents: protst!SNMP-Status
    [2016-05-10 14:40:55 +0200] debug/ServiceDbObject: service contacts: protst!memory-win
    [2016-05-10 14:40:55 +0200] debug/ServiceDbObject: service contactgroups: protst!memory-win
    [2016-05-10 14:40:55 +0200] debug/ServiceDbObject: service dependencies for 'protst!disk-state'
    [2016-05-10 14:40:55 +0200] debug/ServiceDbObject: service parents: protst!SNMP-Status
    [2016-05-10 14:40:55 +0200] debug/ServiceDbObject: service contacts: protst!disk-state
    [2016-05-10 14:40:55 +0200] debug/ServiceDbObject: service contactgroups: protst!disk-state
    [2016-05-10 14:40:55 +0200] debug/ServiceDbObject: service dependencies for 'protst!time'
    [2016-05-10 14:40:55 +0200] debug/ServiceDbObject: service parents: protst!SNMP-Status
    [2016-05-10 14:40:55 +0200] debug/ServiceDbObject: service contacts: protst!time
    [2016-05-10 14:40:55 +0200] debug/ServiceDbObject: service contactgroups: protst!time
    [2016-05-10 14:40:55 +0200] debug/DbEvents: add checkable check history for 'protst!CPU'
    [2016-05-10 14:40:56 +0200] debug/ServiceDbObject: service dependencies for 'protst!SNMP-Status'
    [2016-05-10 14:40:56 +0200] debug/ServiceDbObject: service contacts: protst!SNMP-Status
    [2016-05-10 14:40:56 +0200] debug/ServiceDbObject: service contactgroups: protst!SNMP-Status
    [2016-05-10 14:40:59 +0200] debug/IdoMysqlConnection: Query: UPDATE icinga_hosts SET action_url = '',  active_checks_enabled = '1',  address = '10.139.0.154',  address6 = '',  alias = 'protst',  check_command_args = '',  check_command_object_id = 23,  check_interval = '1',  config_type = '1',  display_name = 'protst',  event_handler_enabled = '1',  failure_prediction_enabled = '0',  flap_detection_enabled = '0',  freshness_checks_enabled = '1',  freshness_threshold = '60',  high_flap_threshold = '30',  host_object_id = 1415,  icon_image = '',  icon_image_alt = '',  instance_id = 1,  low_flap_threshold = '30',  max_check_attempts = '3',  notes = '',  notes_url = '',  notification_interval = '0',  notifications_enabled = '1',  notify_on_down = '0',  notify_on_downtime = '1',  notify_on_flapping = '1',  notify_on_recovery = '1',  notify_on_unreachable = '0',  obsess_over_host = '0',  passive_checks_enabled = '1',  process_performance_data = '1',  retain_nonstatus_information = '1',  retain_status_information = '1',  retry_interval = '1' WHERE host_object_id = 1415
    [2016-05-10 14:40:59 +0200] debug/IdoMysqlConnection: Query: INSERT INTO icinga_scheduleddowntime (author_name, comment_data, downtime_type, duration, endpoint_object_id, entry_time, instance_id, internal_downtime_id, is_fixed, is_in_effect, name, object_id, scheduled_end_time, scheduled_start_time, was_started) VALUES ('pulpy', 'test', '2', '0', 5, FROM_UNIXTIME(1462883988), 1, '651', '1', '1', 'protst!icingam01-p-1462883988-0', 1415, FROM_UNIXTIME(1462884240), FROM_UNIXTIME(1462883940), '1')
    [2016-05-10 14:41:00 +0200] debug/DbEvents: add checkable check history for 'protst!disk-state'
    [2016-05-10 14:41:14 +0200] debug/DbEvents: add checkable check history for 'protst'

Best regards
Mirko

---

**Relations**:

* relates [#11149](https://dev.icinga.com/issues/11149)

**Comment by dnsmichi** [comment]
That is now explicitly mentioned in the docs with #5140.
